Sources/Native/NativeMockServerWrapper.swift

```diff
@@ -2,102 +2,206 @@ import Foundation
-    pactDir = defaultPactDir()
-    port = randomPort()
-  }
-
-  func defaultPactDir() -> String {
-    if let dir = ProcessInfo.processInfo.environment["pact_dir"] {
-      return dir
+#if SWIFT_PACKAGE
+    import SwiftPactServer
+#endif
+
+public class NativeMockServerWrapper: MockServer {
+    private lazy var port: Int32 = {
+        return findUnusedPort()
+    }()
+    private let pactDir: String = ProcessInfo.processInfo.environment["pact_dir"] ?? "/tmp/pacts"
+    private let shouldWritePacts: Bool
+
+    public init(shouldWritePacts: Bool = false) {
+        self.shouldWritePacts = shouldWritePacts
+    }
+
+    private func findUnusedPort() -> Int32 {
+        var port = randomPort
+        var (unused, description) = checkTcpPortForListen(port: port)
+        while !unused {
+            debugPrint(description)
+            port = randomPort
+            (unused, description) = checkTcpPortForListen(port: port)
+        }
+        debugPrint(description)
+        return Int32(port)
+    }
+
+    private var randomPort: in_port_t {
+        return in_port_t(arc4random_uniform(2000) + 4000)
+    }
+
+    public func getBaseUrl() -> String {
+        return "http://localhost:\(port)"
+    }
+
+    public func setup(_ pact: Pact) -> Future<String, PactError> {
+        return Future { complete in
+            do {
+                let payload = pact.payload()
+                let jsonData = try JSONSerialization.data(withJSONObject: payload)
+                let jsonString = String(bytes: jsonData, encoding: String.Encoding.utf8)
+
+                // iOS json generation adds extra backslashes to "application/json" --> "application\\/json"
+                // causing the MockServer to fail to parse the file.
+                let sanitizedString = jsonString!.replacingOccurrences(of: "\\/", with: "/")
+                let result = create_mock_server_ffi(sanitizedString, port)
+                if result < 0 {
+                    switch result {
+                    case -1:
+                        complete(.failure(.setupError("Mock server creation failed, pact supplied was nil")))
+                    case -2:
+                        complete(.failure(.setupError("Mock server creation failed, pact JSON file could not be parsed")))
+                    case -4:
+                        complete(.failure(.setupError("Mock server creation failed, Address already in use")))
+                    default:
+                        complete(.failure(.setupError("Mock server creation failed, result: \(result)")))
+                    }
+                    return
+                }
+                debugPrint("Server started on port \(port)")
+                complete(.success("Server started on port \(port)"))
+            } catch let error as NSError {
+                complete(.failure(.setupError(error.localizedDescription)))
+            }
+        }
+    }
+
+    public func verify(_ pact: Pact) -> Future<String, PactError> {
```

Comment:
You're passing `Pact` into `verify()` yet it is not used in the method.

Reply:
That is true. It's part of the protocol (and the ruby implementation uses it). I could just ignore it here (or move away from the ruby alternative and change the protocol)
TODO: